feat: add header component

biancasoliz-uwu opened the pull request:
Refactor of the new header and change of the routing logic

biancasoliz-uwu committed:
feat: header frontend creation

Design of how the header will look with your navigation.

biancasoliz-uwu committed:
fix: routing logic

The logic of the routes has been changed since the footer and header were always shown, but there are views in which it is not necessary.

biancasoliz-uwu committed:
fix: Header refactor without bootstrap.

header refactored without bootstrap with only html and scss

teran-joseluis commented on src/app/shared/header/header.component.ts:
```diff
@@ -0,0 +1,44 @@
+  }
+  redirectToHome() {
+    this.router.navigate(['/']);
+  }
```
Did you consider the routes of login and sign up?

teran-joseluis replied:
I have this output when I press on both.
![header-logs](https://github.com/FullHouseTeam/pet-paradise-client/assets/73414963/bfe151ff-1ebb-4a5f-9f3c-0885cbc770f5)

biancasoliz-uwu replied:
no, because these are already in the footer with the folder structure.

OropezaHugo commented on src/styles.scss:
```diff
@@ -1,11 +1,166 @@
+}
+.bg-dark{
+  background-color: #4a4950;
+}
```
Why do we need this file to be edited, this is a global scss, and you were suppose to do only 2 components

biancasoliz-uwu replied:
are generic, but I will handle them in the css of the component seeing that they are not useful for the way they work.

teran-joseluis commented on src/app/shared/header/header.component.html:
```diff
@@ -0,0 +1,36 @@
+      <a mat-menu-item href="/login">Login</a>
+    </mat-menu>
+  </div>
+</header>
```
Do not forget to add the empty line please

OropezaHugo commented on src/index.html:
```diff
@@ -8,11 +8,25 @@
+      rel="stylesheet"
+    />
+    <link
+      href="https://fonts.googleapis.com/icon?family=Material+Icons"
```
is this a external library for icons? why didn't you use the angular material library for the icons as it was told?

biancasoliz-uwu replied:
it is not an external library, here I just improved the writing format.

teran-joseluis commented on src/app/shared/header/header.component.html:
```diff
@@ -0,0 +1,36 @@
+  <input type="checkbox" id="menu" hidden>
+  <nav class="d-flex gap-md-5 gap-4 ms-auto align-items-center">
+    <label class="material-icons d-md-none" role="button" for="menu" style="top: 20px; position: fixed;">cancel</label>
+    <a href="">Home</a>
```
Please can you add more spacing between the sections? It looks like they are too close.

OropezaHugo commented on src/app/app.component.ts:
```diff
@@ -1,16 +1,22 @@
+  selector: "pet-paradise-client-root",
+  templateUrl: "./app.component.html",
+  styleUrls: ["./app.component.scss"],
 })
```
we were using the simple cuotes on all the components for all this section, please change it

teran-joseluis commented on src/app/shared/header/header.component.html:
```diff
@@ -0,0 +1,36 @@
+<header class="d-flex gap-3 align-items-center p-3 bg-1-100">
+  <a href="">
+    <img src="assets/logo-black.svg" width="150" alt="logo">
```
Can you add more spacing to the left side of the logo? It seems to be next of the screen.

OropezaHugo commented on src/app/app.component.scss:
```diff
@@ -1,16 +1,23 @@
-  background-color: #CCE8F2;
-  place-items: center;
-
+  align-items: center;
```
this field should be stretch to leave the box go with the content, the way you done it will make the text or content go out from his box in my experience

teran-joseluis commented on src/app/shared/header/header.component.ts:
```diff
@@ -0,0 +1,44 @@
+
+  constructor(private router: Router) {}
+
+  redirectToCategoriesAndProducts() {
```
Why do not you consider using a single method for navigation and pass the route as an argument?

biancasoliz-uwu replied:
sorry, this comes from the previous header, I'm not implementing it like this

teran-joseluis replied:
Have you created a single method? I don't see it.

biancasoliz-uwu replied:
why a method?
I decided to use the HTML a tag. It is used to create links, the href attribute of the a tag specifies the address of the resource to link to.

teran-joseluis commented on src/styles.scss:
```diff
@@ -1,11 +1,166 @@
+}
+.bg-dark{
+  background-color: #4a4950;
+}
```
Do not forget to add the empty line please

OropezaHugo commented on src/app/app.component.ts:
```diff
@@ -1,16 +1,22 @@
+import { HeaderComponent } from "./shared/header/header.component";
+import { HomeComponent } from "../components/home/home.component";
+import { FooterComponent } from "../components/footer/footer.component";
 
```
All this imports should be with simple quotes, not double

OropezaHugo commented on src/app/app.routes.ts:
```diff
@@ -1,22 +1,18 @@
   { path: 'support', component: SupportPageComponent },
   { path: 'terms-and-conditions', component: TermsAndConditionsComponent },
   { path: '', component: HomeComponent },
-  { path: 'example-api', component: DescriptionComponent }
```
please do not erase this route, it is used internally for tests

OropezaHugo commented on src/app/shared/header/header.component.html:
```diff
@@ -0,0 +1,36 @@
+      </mat-accordion>
+    </div>
+  </nav>
+  <a class="mx-md-5 ms-md-5 ms-auto" href="/shop-cart">
```
this large scss code should be in the corresponding scss file, not in the html

biancasoliz-uwu replied:
that's the name, it's not a scss

OropezaHugo commented on src/app/shared/header/header.component.scss:
```diff
@@ -0,0 +1,94 @@
+  color: #f6cd2a !important;
+}
+
+@media (min-width: 768px) {
```
instead of make media for responsive, should focus on our target resolutions for mobile

biancasoliz-uwu replied:
2 designs are handled so I needed this consideration, it was not to make responsive media

biancasoliz-uwu committed:
fix: corrected header based on comments

header scss styles are handled inside the component, single quotes handling, removal of repeated icon library

biancasoliz-uwu committed:
fix: improved header styling

Improved sections and logo, removed unused redirection, handled missing pages and added spaces at the end of classes.

biancasoliz-uwu committed:
fix: update of drop-down menu background in mobile view
